sc-kpm/sc-python/services/http_api/auth/database.py

```diff
@@ -0,0 +1,91 @@
+from sqlalchemy import Column, Integer, String, ForeignKey
+from sqlalchemy.engine import Engine
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.declarative import declarative_base
+
+from sqlalchemy import create_engine, event
+from sqlalchemy.orm import sessionmaker, relationship
+
+import http_api.auth.constants as cnt
+from http_api.auth.config import params
+
+Base = declarative_base()
+
+
+@event.listens_for(Engine, "connect")
+def set_sqlite_pragma(dbapi_connection, connection_record):
+    cursor = dbapi_connection.cursor()
+    cursor.execute("PRAGMA foreign_keys=ON")
+    cursor.close()
+
+
+class User(Base):
+    __tablename__ = cnt.USER
+    id = Column(Integer, primary_key=True, unique=True)
+    name = Column(String(255), nullable=False)
+    password = Column(String(255), nullable=False)
+
+    def __repr__(self):
+        return '<id={}, name={}>'.format(self.id, self.name)
+
+    @property
+    def serialize(self):
+        return {
+            cnt.ID: self.id,
+            cnt.NAME: self.name
+        }
+
+class DataBase:
+    def __init__(self) -> None:
+        self.engine = create_engine(params[cnt.SQLITE_DB_PATH])
+        self.session = None
+
+    def init(self) -> None:
+        Base.metadata.create_all(self.engine)
+        self._session().commit()
+
+    def _session(self):
+        if not self.session:
+            self.session = sessionmaker(bind=self.engine)()
+        return self.session
+
+    def is_user_valid(self, name, password):
+        selected_user = self._session().query(User).filter(User.name == name, User.password == password).first()
+        return selected_user is not None
+
+    def is_such_user_in_base(self, name):
+        selected_user = self._session().query(User).filter(User.name == name).first()
+        return selected_user is not None
+
+    def get_user_by_id(self, id: int):
+        user_info = self._session().query(User).filter(User.id == id).first()
+        return user_info.serialize if user_info is not None else None
+
+    def get_users(self):
+        users_info = self._session().query(User)
+        users_info = [x.serialize for x in users_info.all()]
+        return users_info
+
+    def update_user(self, u):
+        self._session().merge(u)
+        self._session().commit()
```

Comment:
Why we need it if we have update by id? Also we dont use this function anywhere.

Reply:
Yes, it's legacy code from sc-web. May be deleted.